lib/message_bus/message.rb

```diff
@@ -9,8 +9,12 @@ def self.decode(encoded)
     s2 = encoded.index("|", s1 + 1)
     s3 = encoded.index("|", s2 + 1)
 
-    MessageBus::Message.new(encoded[0..s1].to_i, encoded[(s1 + 1)..s2].to_i,
-                            encoded[(s2 + 1)..(s3 - 1)].gsub("$$123$$", "|"), encoded[(s3 + 1)..-1])
+    global_id  = encoded[0, s1+1].to_i
+    message_id = encoded[(s1 + 1), (s2 - s1 - 1)].to_i
+    channel    = encoded[(s2 + 1), (s3 - s2 - 1)].gsub("$$123$$", "|")
```

Comment:
cool, do we just gsub! here? one less string alloc.

Reply:
@SamSaffron `gsub!` returns nil if there's no match, whereas `gsub` returns the original string. So, it doesn't seem correct (also it breaks a few tests).

Reply:
Sorry I mean:

```
channel    = encoded[(s2 + 1), (s3 - s2 - 1)]
channel.gsub!("$$123$$", "|")
```

Its one less string alloc ... calling gsub when there is no hit is the same as calling dup, one extra string